documentation/src/main/asciidoc/user_guide/marshalling.adoc

```diff
@@ -13,13 +13,13 @@ Since performance is a very important factor in this process, Infinispan uses JB
 === Support For Non-Serializable Objects
 From a users perspective, a very common concern is whether Infinispan supports storing non-Serializable objects. In 4.0, an Infinispan cache instance can only store non-Serializable key or value objects if, and only if:
 
-*  cache is configured to be a local cache _and..._ 
-*  cache is not configured with lazy serialization _and..._ 
+*  cache is configured to be a local cache _and..._
+*  cache is not configured with lazy serialization _and..._
 * cache is not configured with any write-behind cache store
 
 If either of these options is true, key/value pairs in the cache will need to be marshalled and currently they require to either to extend `java.io.Serializable` or `java.io.Externalizable`.
 
-TIP: Since Infinispan 5.0, marshalling non-Serializable key/value objects is supported as long as users can to provide meaningful Externalizer implementations for these non-Seralizable objects. <<_plugging_infinispan_with_user_defined_externalizers, This section>> has more details. 
+TIP: Since Infinispan 5.0, marshalling non-Serializable key/value objects is supported as long as users can to provide meaningful Externalizer implementations for these non-Seralizable objects. <<_plugging_infinispan_with_user_defined_externalizers, This section>> has more details.
 
 If you're unable to retrofit Serializable or Externalizable into the classes whose instances are stored in Infinispan, you could alternatively use something like link:http://x-stream.github.io/[XStream] to convert your Non-Serializable objects into a String that can be stored into Infinispan. You can find an example on how to use XStream link:http://anonsvn.jboss.org/repos/infinispan/trunk/core/src/test/java/org/infinispan/marshall/TestObjectStreamMarshaller.java[here] . The one caveat about using XStream is that it slows down the process of storing key/value objects due to the XML transformation that it needs to do.
```

Comment:
This link is really old, and the newest `TestObjectStreamMarshaller` doesn't use XStream any more...